packages/explorer/src/observer/store.ts

```diff
@@ -10,11 +10,13 @@ export type Write = {
   writeId: string;
   type: MessageType;
   hash?: Hex;
+  from?: Address;
   address: Address;
   functionSignature: string;
   args: unknown[];
   time: number;
   events: Message<Exclude<MessageType, "ping">>[];
+  error?: Error;
 };
```

Comment:
just FYI that we should be aware of changes here because it's not guaranteed that the client sending data to the explorer are version-aligned, so although explorer makes some assumptions about new data here, an older client may not be passing it over yet

this currently would only show up if you configured your observer to point to the hosted explorer rather than a local one

Reply:
that said, doing this as an optional type is a good way to safe-guard against this!

Reply:
yeah great point, will keep this in mind going forward. For now all the new fields are optional so we are safe there

Reply:
actually my mistake, this is the internal store, which is fine to extend!

the messages definition + decorator is the thing to be mindful of